discord/guild.py

```diff
@@ -764,6 +766,22 @@ def get_emoji(self, emoji_id: int, /) -> Optional[Emoji]:
         if emoji and emoji.guild == self:
             return emoji
         return None
+    
+    def get_guild_shop(self) -> Optional[GuildShop]:
```

Comment:
```suggestion
    async def guild_shop(self) -> GuildShop:
```


Methods which make a HTTP request for data shouldn't use the `get_` prefix.

Also, I'm assuming discord returns a 404 when there is no guild shop, you should propagate that rather than wrapping returning an `Optional` type.

Also, since this is fetching from the API, this needs to be an `async` function.

Reply:
Ok, changed the name from get_guild_shop to guild_shop and made it asynchronous.